### ISSUE TITLE
[BUG] @SuperBuilder: Name clash when using 

### ISSUE THREAD

**Issue by PhoenixmitX** [projectlombok/lombok#3295]
**Describe the bug**
the following code does not compile

**To Reproduce**
```java
import lombok.experimental.SuperBuilder;

public class SuperBuilderError {
	
	@SuperBuilder
	static class Foo<T extends Foo<T>> {
	}

	@SuperBuilder // Name clash: The method builder() of type SuperBuilderError.Bar<T> has the same erasure as builder() of type SuperBuilderError.Foo<T> but does not hide it - Java(67109448)
	static class Bar<T extends Bar<T>> extends Foo<T> {
	}
}
```

i only put it in a subclass to have a small copyable snipet
it also does not compile if you put it into different files

**Expected behavior**
well compile correctly

**Version info (please complete the following information):**
 - Lombok version: 1.18.24
 - javac 17.0.3
 - VSCode 1.73.0
 - gradle 7.5
 - sourceCompatibility = 8
 - targetCompatibility = 8

**Additional context**
		`¯\_(ツ)_/¯`

**Comment by janrieke** [comment]
Can confirm, but I don't think this is fixable. Similar to #2524, it's probably beyond the limits of Java generics.

Here's the delombok'ed code. If you can find a type-safe way to make this compile, I'd be happy to try to modify the generated code. I haven't found one.

	public class SuperBuilderErrorDelombok {
		static class Foo<T extends Foo<T>> {
			public static abstract class FooBuilder<T extends Foo<T>, C extends SuperBuilderErrorDelombok.Foo<T>, B extends SuperBuilderErrorDelombok.Foo.FooBuilder<T, C, B>> {
				protected abstract B self();
	
				public abstract C build();
	
				@java.lang.Override
				public java.lang.String toString() {
					return "SuperBuilderErrorDelombok.Foo.FooBuilder()";
				}
			}
	
			private static final class FooBuilderImpl<T extends Foo<T>> extends
					SuperBuilderErrorDelombok.Foo.FooBuilder<T, SuperBuilderErrorDelombok.Foo<T>, SuperBuilderErrorDelombok.Foo.FooBuilderImpl<T>> {
				private FooBuilderImpl() {
				}
	
				@java.lang.Override
				protected SuperBuilderErrorDelombok.Foo.FooBuilderImpl<T> self() {
					return this;
				}
	
				@java.lang.Override
				public SuperBuilderErrorDelombok.Foo<T> build() {
					return new SuperBuilderErrorDelombok.Foo<T>(this);
				}
			}
	
			protected Foo(final SuperBuilderErrorDelombok.Foo.FooBuilder<T, ?, ?> b) {
			}
	
			public static <T extends Foo<T>> SuperBuilderErrorDelombok.Foo.FooBuilder<T, ?, ?> builder() {
				return new SuperBuilderErrorDelombok.Foo.FooBuilderImpl<T>();
			}
		}
	
		// Name clash: The method builder() of type SuperBuilderErrorDelombok.Bar<T> has
		// the same erasure as builder() of type SuperBuilderErrorDelombok.Foo<T> but
		// does not hide it - Java(67109448)
		static class Bar<T extends Bar<T>> extends Foo<T> {
	
			public static abstract class BarBuilder<T extends Bar<T>, C extends SuperBuilderErrorDelombok.Bar<T>, B extends SuperBuilderErrorDelombok.Bar.BarBuilder<T, C, B>>
					extends Foo.FooBuilder<T, C, B> {
				@java.lang.Override
				protected abstract B self();
	
				@java.lang.Override
				public abstract C build();
	
				@java.lang.Override
				public java.lang.String toString() {
					return "SuperBuilderErrorDelombok.Bar.BarBuilder(super=" + super.toString() + ")";
				}
			}
	
			private static final class BarBuilderImpl<T extends Bar<T>> extends
					SuperBuilderErrorDelombok.Bar.BarBuilder<T, SuperBuilderErrorDelombok.Bar<T>, SuperBuilderErrorDelombok.Bar.BarBuilderImpl<T>> {
	
				private BarBuilderImpl() {
				}
	
				@java.lang.Override
				protected SuperBuilderErrorDelombok.Bar.BarBuilderImpl<T> self() {
					return this;
				}
	
				@java.lang.Override
				public SuperBuilderErrorDelombok.Bar<T> build() {
					return new SuperBuilderErrorDelombok.Bar<T>(this);
				}
			}
	
			protected Bar(final SuperBuilderErrorDelombok.Bar.BarBuilder<T, ?, ?> b) {
				super(b);
			}
	
			public static <T extends Bar<T>> SuperBuilderErrorDelombok.Bar.BarBuilder<T, SuperBuilderErrorDelombok.Bar<T>, SuperBuilderErrorDelombok.Bar.BarBuilderImpl<T>> builder() {
				return new SuperBuilderErrorDelombok.Bar.BarBuilderImpl<T>();
			}
		}
	}

**Comment by PhoenixmitX** [comment]
I didn't delomboked the code.
Now i understand the problem ty.
I think its unsolvable too.

But i have two improvements that can be made:
- Wirte a custom ErrorMessage on the annotation.
- add a option to hide the builder() method - in my case i don't need it on the root class, but i have multiple subclasses that all need a builder() method

**Comment by janrieke** [comment]
I did not test it, but IIRC you can simply use `@SuperBuilder(builderMethodName="")` to suppress the generation of the `builder` method.

**Comment by PhoenixmitX** [comment]
You're right but i think its not documented and a boolean value would be a cleaner implementation

**Comment by PhoenixmitX** [comment]
something like suppressBuilderMethod = true
